pimd/pim_iface.c

```diff
@@ -209,6 +209,7 @@ void pim_if_delete(struct interface *ifp)
 	list_delete(&pim_ifp->pim_neighbor_list);
 	list_delete(&pim_ifp->upstream_switch_list);
 	list_delete(&pim_ifp->sec_addr_list);
+    list_delete(&pim_ifp->gm_join_list);
```

Comment:
Check line 192 to 196. You need to make that function common to both pimv4 and pimv6. 
You just need to remove the #if and rename the api pim_if_igmp_join_del_all to pim_if_gm_join_del_all

Reply:
Thank you for the valuable input!